third_party/maya/lib/usdMaya/JobArgs.cpp

```diff
@@ -417,6 +423,44 @@ const VtDictionary& JobExportArgs::GetDefaultDictionary()
     return d;
 }
 
+void JobExportArgs::addFilteredTypeName(const MString& typeName)
+{
+    MNodeClass cls(typeName);
+    unsigned int id = cls.typeId().id();
+    if (id == 0) {
+        MGlobal::displayWarning(MString("Given excluded node type '") + typeName
+                + "' does not exist; ignoring");
+        return;
+    }
+    filteredTypeIds.insert(id);
+    // We also insert all inherited types - only way to query this is through mel,
+    // which is slower, but this should be ok, as these queries are only done
+    // "up front" when the export starts, not per-node
+    MString queryCommand("nodeType -isTypeName -derived ");
+    queryCommand += typeName;
+    MStringArray inheritedTypes;
+    MStatus status = MGlobal::executeCommand(queryCommand, inheritedTypes, false, false);
+    if (!status) {
+        MGlobal::displayWarning(MString("Error querying derived types for '") + typeName
+                + "': " + status.errorString());
+        return;
+    }
+
+    for (unsigned int i=0; i < inheritedTypes.length(); ++i) {
+        if (inheritedTypes[i].length() == 0) continue;
+        id = MNodeClass(inheritedTypes[i]).typeId().id();
+        if (id == 0) {
+            // Unfortunately, the returned list will often include weird garbage, like
+            // "THconstraint" for "constraint", which cannot be converted to a MNodeClass,
+            // so just ignore these...
+            // MGlobal::displayError(MString("Given inherited excluded node type '") + inheritedTypes[i]
```

Comment:
Is keeping this commented-out displayError call intentional?

Reply:
It was - I liked that it was sort of descriptive of what was going on, but it's not really necessary. I'll get rid of it.